### PR TITLE
Billing: Update billing-history components to use Redux

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates `BillingHistory` and `BillingReceipt` to use the new `billing-transactions` Redux functionality. It also adds a simple placeholder to `BillingReceipt`, which was borrowed from the one in `BillingHistory` - this will show a `Loading...` message in the rare cases when the transaction is not loaded. 

This PR is part of #10554.

To test:

* Checkout this branch
* Go to `/me/purchases/billing` and verify purchases are loaded properly. There should be no functional changes.
* Click on "View Receipt" link in a purchase of your choice.
* Verify the receipt page works like before, and it displays all transaction data properly.
* Try the above with a clean Redux state and verify everything works as expected.

### CODE REVIEW

File: client/me/billing-history/receipt.jsx [hunk 1]
```diff
@@ -126,6 +130,36 @@ const BillingReceipt = React.createClass( {
 		);
 	},
 
+	renderTitle() {
+		const { translate } = this.props;
+
+		return (
+			<div>
+				<QueryBillingTransactions />
```
Comment (gwwar) [inline client/me/billing-history/receipt.jsx]: We'll always want to render the query component on mount, and make data needs as clear as possible. How about something similar to:
```javascript
render() {
    const { transaction } = this.props;
    return(
        <Main>
            <QueryBillingTransactions />
            { this.renderTitle() }
            { ! transaction ? this.renderPlaceholder() : this.renderBillingHistory() }
       </Main>
   );
}
```

File: client/me/billing-history/receipt.jsx [hunk 2]
```diff
@@ -126,6 +130,36 @@ const BillingReceipt = React.createClass( {
...
+	renderPlaceholder() {
+		const { translate } = this.props;
+
+		return (
+			<Main>
+				{ this.renderTitle() }
+				<Card compact className="billing-history__receipt-card">
+					<div className="billing-history__receipt">
+						<div className="billing-history__receipt-loading">
+							{ translate( 'Loading…' ) }
```
Comment (gwwar) [inline client/me/billing-history/receipt.jsx]: This works, though we tend to use a pulsing block loader. (cc @mtias for suggestions) We have a sass [mixin](https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/shared/mixins/_placeholder.scss) for this, which you can add to your styles with: 
```scss
.component__placeholder {
     @include placeholder();
}
```

Feel free to update this in a follow up PR though, as prod currently breaks in this receipt view with a clean cache (localStorage cleared).

File: client/me/billing-history/controller.js [hunk 3]
```diff
@@ -19,13 +19,12 @@ const sites = sitesFactory();
 export default {
 	billingHistory( context ) {
 		const BillingHistoryComponent = require( './main' );
-		const billingData = require( 'lib/billing-history-data' );
```
Comment (tyxla) [inline client/me/billing-history/controller.js]: Of course, I was planning to do so in a separate PR and this is one of the outstanding tasks in #10554.

File: client/me/billing-history/controller.js [hunk 4]
```diff
@@ -35,20 +34,16 @@ export default {
...
 		const receiptId = context.params.receiptId;
 		const basePath = route.sectionify( context.path );
 
-		// Initialize billing data
-		billingData.get();
-
 		context.store.dispatch( setTitle( i18n.translate( 'Billing History', { textOnly: true } ) ) ); // FIXME: Auto-converted from the Flux setTitle action. Please use <DocumentHead> instead.
 
 		if ( receiptId ) {
 			analytics.pageView.record( basePath + '/receipt', ANALYTICS_PAGE_TITLE + ' > Billing History > Receipt' );
```
Comment (tyxla) [inline client/me/billing-history/controller.js]: Good point, I'm adding this to #10554 and will address it in another PR.

File: client/me/billing-history/main.jsx [hunk 5]
```diff
@@ -19,31 +19,32 @@ import BillingHistoryTable from './billing-history-table';
...
+		const { transactions, sites, translate } = this.props;
 
 		return (
 			<Main className="billing-history">
 				<MeSidebarNavigation />
+				<QueryBillingTransactions />
 				<PurchasesHeader section={ 'billing' } />
 				<Card className="billing-history__receipts">
-					<BillingHistoryTable transactions={ data.billingHistory } />
+					<BillingHistoryTable transactions={ transactions.past } />
```
Comment (tyxla) [inline client/me/billing-history/main.jsx]: Makes sense, I'm adding this to #10554 and will address it in another PR.

File: client/me/billing-history/receipt.jsx (hunk 1 above)
Comment (tyxla) [inline client/me/billing-history/receipt.jsx]: Looks better this way, but I'll leave the refactor for a next PR. Adding to #10554.

File: client/me/billing-history/receipt.jsx (hunk 2 above)
Comment (tyxla) [inline client/me/billing-history/receipt.jsx]: That would work great - and I'll probably consider doing that for both billing history list and receipt pages for consistency. Adding to #10554.

Comment (mtias) [inline client/me/billing-history/receipt.jsx]: Agreed with using placeholders instead of loading... text here.
